ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/commons/RaoUtil.java

```diff
@@ -225,4 +237,99 @@ public static Set<String> getDuplicateCnecs(Set<FlowCnec> flowcnecs) {
             .map(FlowCnec::getId)
             .collect(Collectors.toSet());
     }
+
+    /**
+     * Add to the crac a network action that deactivate ac emulation for each hvdc line in ac emulation mode that has at least one hvdc range action associated
+     */
+    public static void addNetworkActionAssociatedWithHvdcRangeAction(Crac crac, Network network) {
+        crac.getHvdcRangeActions().forEach(hvdcRangeAction -> {
```

Comment:
could you add step by step comments ? 
// Fetch hvdc id
// check if a NA has been defined
// if not defined
// if alreay defined for anaother range action, add current range action's usage rules